variation-commons-mongodb/src/main/java/uk/ac/ebi/eva/commons/mongodb/services/VariantSourceService.java

```diff
@@ -44,4 +44,7 @@ public long countByFileIdIn(List<String> fileIds) {
         return repository.countByFileIdIn(fileIds);
     }
 
+    public List<VariantSourceMongo> findAllVariantSourcesForBeacon() {
```

Comment:
It retrieves only the required fields. I thought it would be more efficient.
Should I change it?

`    @Query(value = "{}" , fields = "{sid:1,sname:1,date:1,st:1}")
    List<VariantSourceMongo> findAllForBeacon();`

Reply:
That is true, returning all the fields when you are only going to need some is inefficient. Would it be possible to wrap it with a call to `convert`, as in `findAll`?